src/anemoi/models/models/encoder_processor_decoder.py

```diff
@@ -202,32 +199,41 @@ def _run_mapper(
         )
 
     def forward(self, x: Tensor, model_comm_group: Optional[ProcessGroup] = None) -> Tensor:
-        batch_size = x.shape[0]
-        ensemble_size = x.shape[2]
+        batch_size, _, ensemble_size, *_ = x.shape
```

Comment:
Isn't this much less readable?

Reply:
For me it's more visual, but I understand that it's a very personal thing. I leave it as it was